"Cannot get property 'extid' on null object" when viewing job definition

**Describe the bug**
When clicking on a specific job through the UI, I get an error as stated below. This only happens on a specific job though I've not done a thorough search of other jobs this may effect. Any jobs linked to this job show the same error.

**My Rundeck detail**

* Rundeck version: 2.11.3
* RPM via Chef cokobook
* OS Name/version: CentOS 7
* DB Type/version: MySQL 5.6.35

**To Reproduce**
Steps to reproduce the behavior:
1. Sign into Rundeck
2. Click on project (Helpers)
3. Click on job to view definition
4. Error: Error processing GroovyPageView: Error executing tag <g:render>: Error executing tag <g:render>: Error executing tag <g:render>: Error executing tag <auth:resourceAllowed>: Error executing tag <g:form>: Cannot get property 'extid' on null object

```
Stacktrace:
ERROR GrailsExceptionResolver: NullPointerException occurred when processing request: [GET] /project/Helpers/job/show/b0b9ef5d-cadf-4d21-b4a2-590880c0cc40
Cannot get property 'extid' on null object. Stacktrace follows:
org.codehaus.groovy.grails.web.pages.exceptions.GroovyPagesException: Error processing GroovyPageView: Error executing tag <g:render>: Error executing tag <g:render>: Error executing tag <g:render>: Error executing tag <auth:resourceAllowed>: Error executing tag <g:form>: Cannot get property 'extid' on null object
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:696)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1515)
        at grails.plugin.cache.web.filter.PageFragmentCachingFilter.doFilter(PageFragmentCachingFilter.java:198)
        at grails.plugin.cache.web.filter.AbstractFilter.doFilter(AbstractFilter.java:63)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:519)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:138)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:582)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:213)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1097)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:448)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:175)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1031)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:136)
        at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:261)
        at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:101)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at com.codahale.metrics.servlet.AbstractInstrumentedFilter.doFilter(AbstractInstrumentedFilter.java:97)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at com.dtolabs.rundeck.server.filters.AuthFilter.doFilter(AuthFilter.java:74)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:519)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:138)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:529)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:213)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1097)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:448)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:175)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1031)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:136)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
        at org.eclipse.jetty.server.Server.handle(Server.java:446)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:271)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:246)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.run(AbstractConnection.java:358)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:601)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:532)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.codehaus.groovy.grails.web.taglib.exceptions.GrailsTagException: Error executing tag <g:render>: Error executing tag <g:render>: Error executing tag <g:render>: Error executing tag <auth:resourceAllowed>: Error executing tag <g:form>: Cannot get property 'extid' on null object
        at gsp_rundeck_scheduledExecutionshow_gsp$_run_closure2.doCall(gsp_rundeck_scheduledExecutionshow_gsp.groovy:68)
        at gsp_rundeck_scheduledExecutionshow_gsp.run(gsp_rundeck_scheduledExecutionshow_gsp.groovy:73)
        ... 44 more
Caused by: org.codehaus.groovy.grails.web.taglib.exceptions.GrailsTagException: Error executing tag <g:render>: Error executing tag <g:render>: Error executing tag <auth:resourceAllowed>: Error executing tag <g:form>: Cannot get property 'extid' on null object
        at gsp_rundeck_scheduledExecution_show_gsp.run(gsp_rundeck_scheduledExecution_show_gsp.groovy:19)
        ... 46 more
Caused by: org.codehaus.groovy.grails.web.taglib.exceptions.GrailsTagException: Error executing tag <g:render>: Error executing tag <auth:resourceAllowed>: Error executing tag <g:form>: Cannot get property 'extid' on null object
        at gsp_rundeck_scheduledExecution_showHead_gsp.run(gsp_rundeck_scheduledExecution_showHead_gsp.groovy:119)
        ... 47 more
Caused by: org.codehaus.groovy.grails.web.taglib.exceptions.GrailsTagException: Error executing tag <auth:resourceAllowed>: Error executing tag <g:form>: Cannot get property 'extid' on null object
        at gsp_rundeck_scheduledExecution_jobActionButton_gsp.run(gsp_rundeck_scheduledExecution_jobActionButton_gsp.groovy:214)
        ... 48 more
Caused by: org.codehaus.groovy.grails.web.taglib.exceptions.GrailsTagException: Error executing tag <g:form>: Cannot get property 'extid' on null object
        at gsp_rundeck_scheduledExecution_jobActionButton_gsp$_run_closure3.doCall(gsp_rundeck_scheduledExecution_jobActionButton_gsp.groovy:209)
        at rundeck.AuthTagLib$_closure3.doCall(AuthTagLib.groovy:123)
        ... 49 more
Caused by: java.lang.NullPointerException: Cannot get property 'extid' on null object
        at gsp_rundeck_scheduledExecution_jobActionButton_gsp$_run_closure3$_closure8.doCall(gsp_rundeck_scheduledExecution_jobActionButton_gsp.groovy:196)
        at rundeck.UtilityTagLib$_closure51.doCall(UtilityTagLib.groovy:1100)
        ... 51 more
```

**Expected behavior**
View job definition in UI


**Desktop (please complete the following information):**
 - OS: MacOS, Windows
 - Browser: Chrome
 - Version: 66.0.3359.181

**Additional context**
Recently upgraded from 2.10.x to 2.11.3

--- a/rundeckapp/grails-app/domain/rundeck/ReferencedExecution.groovy
+++ b/rundeckapp/grails-app/domain/rundeck/ReferencedExecution.groovy
@@ -17,7 +17,7 @@ class ReferencedExecution {
     static List<ScheduledExecution> parentList(ScheduledExecution se, int max = 0){
         def refExecs = findAllByScheduledExecution(se)?.collect{ re ->
             re.execution?.scheduledExecution
-        }?.unique()
+        }?.unique().findAll(){it!=null}
         refExecs.subList(0, (max!=0 && refExecs.size()>max)?max:refExecs.size())
 
     }

--- a/rundeckapp/test/unit/rundeck/controllers/ScheduledExecutionControllerSpec.groovy
+++ b/rundeckapp/test/unit/rundeck/controllers/ScheduledExecutionControllerSpec.groovy
@@ -1707,4 +1707,99 @@ class ScheduledExecutionControllerSpec extends Specification {
         1 * controller.executionService.respondExecutionsXml(_, _, _)
         0 * controller.executionService._(*_)
     }
+
+        def "isReference deleted parent"(){
+        given:
+        def refTotal = 10
+        def se = new ScheduledExecution(
+                uuid: jobuuid,
+                jobName: 'test1',
+                project: 'project1',
+                groupPath: 'testgroup',
+                doNodedispatch: true,
+                filter:'name: ${option.nodes}',
+                refExecCount: refTotal,
+                workflow: new Workflow(
+                        keepgoing: true,
+                        commands: [
+                                new CommandExec([
+                                        adhocRemoteString: 'test buddy',
+                                        argString: '-delay 12 -monkey cheese -particle'
+                                ])
+                        ]
+                )
+        ).save()
+        def seb = new ScheduledExecution(
+                jobName: 'test2',
+                project: 'project1',
+                groupPath: 'testgroup',
+                doNodedispatch: true,
+                filter:'name: ${option.nodes}',
+                refExecCount: refTotal,
+                workflow: new Workflow(
+                        keepgoing: true,
+                        commands: [
+                                new CommandExec([
+                                        adhocRemoteString: 'test buddy',
+                                        argString: '-delay 12 -monkey cheese -particle'
+                                ])
+                        ]
+                )
+        ).save()
+        def exec = new Execution(
+                user: "testuser",
+                project: "project1",
+                loglevel: 'WARN',
+                status: 'FAILED',
+                doNodedispatch: true,
+                filter:'name: nodea',
+                succeededNodeList:'fwnode',
+                failedNodeList: 'nodec xyz,nodea',
+                workflow: new Workflow(commands: [new CommandExec(adhocExecution: true, adhocRemoteString: 'a remote string')]).save(),
+                scheduledExecution: null
+        ).save()
+
+        //if(expected){
+            def re = new ReferencedExecution(scheduledExecution: se,execution: exec).save()
+        //}
+
+
+
+        NodeSetImpl testNodeSet = new NodeSetImpl()
+        testNodeSet.putNode(new NodeEntryImpl("nodea"))
+        testNodeSet.putNode(new NodeEntryImpl("nodeb"))
+        testNodeSet.putNode(new NodeEntryImpl("nodec xyz"))
+        NodeSetImpl testNodeSetB = new NodeSetImpl()
+        testNodeSetB.putNode(new NodeEntryImpl("nodea"))
+        testNodeSetB.putNode(new NodeEntryImpl("nodec xyz"))
+
+        controller.frameworkService=Mock(FrameworkService){
+            authorizeProjectJobAny(_,_,_,_)>>true
+            filterAuthorizedNodes(_,_,_,_)>>{args-> args[2]}
+            filterNodeSet(_,_)>>testNodeSetB
+            getRundeckFramework()>>Mock(Framework){
+                getFrameworkNodeName()>>'fwnode'
+            }
+        }
+        controller.scheduledExecutionService=Mock(ScheduledExecutionService){
+            getByIDorUUID(_)>>se
+        }
+        controller.notificationService=Mock(NotificationService)
+        controller.orchestratorPluginService=Mock(OrchestratorPluginService)
+        controller.pluginService = Mock(PluginService)
+        when:
+        request.parameters = [id: se.id.toString(),project:'project1',retryFailedExecId:exec.id.toString()]
+
+        def model = controller.show()
+        then:
+        response.redirectedUrl==null
+        model != null
+        model.isReferenced==false
+
+        where:
+        jobuuid     | expected
+        '000000'    | false
+        '111111'    | false
+
+    }
 }

